CRM/Event/Form/Registration.php

```diff
@@ -56,7 +58,14 @@ protected function getOrder(): CRM_Financial_BAO_Order {
    */
   public function getEventID(): int {
     if (!$this->_eventId) {
-      $this->_eventId = (int) CRM_Utils_Request::retrieve('id', 'Positive', $this, TRUE);
+      try {
+        $this->_eventId = (int) CRM_Utils_Request::retrieve('id', 'Positive', $this, TRUE);
+      }
+      catch (CRM_Core_Exception $e) {
```

Comment:
I would propose creating a new 'CRM_Utils_System::sendNotFoundResponse(ts('Missing Event ID')` - just so that `use GuzzleHttp\Psr7\Response;` can be limited to one file so that when we decide to change libraries (!) it is easier